sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/PercentEscaper.java

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.authorization.implementation;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.Arrays;
+
+/**
+ * An escaper that escapes URL data through percent encoding.
+ */
+public final class PercentEscaper {
```

Comment:
copied from azure-core

Reply:
Seems it's fine to use '+' to replace space in query parameters? https://www.w3.org/TR/html4/interact/forms.html#h-17.13.4.1

Is it service that doesn't support this kind of encoding?

Reply:
that is for `application/x-www-form-urlencoded`, not sure if it applies for query parameter.

I think service likely can accept "+". But my concern is that it would be safer if we do same as azure-core does.